client/src/main/java/io/pravega/client/stream/impl/TransactionalEventStreamWriterImpl.java

```diff
@@ -145,15 +144,15 @@ public void abort() {
                         log.debug("Got exception while writing to transaction on abort: {}", e.getMessage());
                     }
                 }
-                getAndHandleExceptions(controller.abortTransaction(stream, txId), RuntimeException::new);
+                Futures.getThrowingException(controller.abortTransaction(stream, txId));
                 closed.set(true);
             }
         }
 
         @Override
         public Status checkStatus() {
             log.info("Check transaction status {}", txId);
-            return getAndHandleExceptions(controller.checkTransactionStatus(stream, txId), RuntimeException::new);
+            return Futures.getThrowingException(controller.checkTransactionStatus(stream, txId));
```

Comment:
checkTransactionStatus will never throw a TxnFailedException ; as I understand the intent here is to remove redundant RuntimeException wrapping.

Reply:
Yes Sandeep